### PR TITLE
added a llamaIndex ColBERT retriever

### PR DESCRIPTION
**Pull request by epinzur**
this is written for LlamaIndex > 0.10.0 which isn't in ragstack yet :(

### CODE REVIEW

File: ragstack/colbert/llamaindex/retriever.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        similarity_top_k: int = DEFAULT_SIMILARITY_TOP_K,
+        callback_manager: Optional[CallbackManager] = None,
+        object_map: Optional[dict] = None,
+        verbose: bool = False,
+        query_maxlen: int = -1,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        self._retriever = retriever
+        self._vector_store = self._index.vector_store
```
Comment (zzzming) [inline ragstack/colbert/llamaindex/retriever.py:32]: where should _index be from

File: ragstack/colbert/llamaindex/retriever.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        similarity_top_k (int): number of top k results to return.
+    """
+
+    def __init__(
+        self,
+        retriever: ColbertVectorStoreRetriever,
+        similarity_top_k: int = DEFAULT_SIMILARITY_TOP_K,
+        callback_manager: Optional[CallbackManager] = None,
+        object_map: Optional[dict] = None,
+        verbose: bool = False,
```
Comment (zzzming) [inline ragstack/colbert/llamaindex/retriever.py:26]: Is verbose required for Ilama for Colbert? It's an int in Colbert. But I saw LlamaIndex has verbose too

**Review by zzzming — APPROVED** [review]

File: ragstack/colbert/llamaindex/retriever.py (hunk 2 above)
Comment (epinzur) [reply ragstack/colbert/llamaindex/retriever.py:26]: yes it is part of the BaseRetriever:

https://github.com/run-llama/llama_index/blob/main/llama-index-core/llama_index/core/base/base_retriever.py#L50

File: ragstack/colbert/llamaindex/retriever.py (hunk 1 above)
Comment (epinzur) [reply ragstack/colbert/llamaindex/retriever.py:32]: good catch, dropping this.

**Pull request merged**